src/freenet/client/filter/M3UFilter.java

```diff
@@ -164,7 +168,7 @@ public void readFilter(
                             }
                             // write the newline if we're not at EOF
                             if (readcount != -1){
-                                dos.write(nextbyte);
+                                dos.write(System.lineSeparator().getBytes(StandardCharsets.UTF_8));
```

Comment:
This change makes the filter’s behaviour dependent on the OS Fred is being run on. Is that actually what we want? In my opinion our filters should behave the same everywhere, especially as the output of the filter is supposed to be usable on every OS. Now, we of course do not know here what OS the output of the filter is being used on so really any choice we make here is an arbitrary one, which is a rather long-winded way of saying that we shouldn’t actually change anything here. 🙂

Reply:
I think the likelyhood that we’ll use the file on Windows if the filter runs on Windows is pretty high. And there are still editors on Windows that display a file with `\n` as line separator as just one long line.

I don’t think any audio player makes that difference, though, so this is rather about people reading the code, not about functionality.

Reply:
True, people using notepad.exe would probably prefer Windows-style line endings… 😁 

But, this would also expose that the node is running on a Windows system. I’m not sure how much of an issue _that_ is, though… are we worrying about that?